src/npc.h

```diff
@@ -79,6 +79,12 @@ using overmap_location_str_id = string_id<overmap_location>;
 using drop_location = std::pair<item_location, int>;
 using drop_locations = std::list<drop_location>;
 
+// This function is used to convert the tags </color> and <color_XXX> into[/color] and [color_XXX]
+// and restore the color tags during tag parsing.
+// In the parse_tags function, the color tag affects the parsing of other <> forms.
+// To avoid this issue, it is necessary to temporarily remove it.
+void replace_color_tags( std::string &str, std::string from_tag_left, std::string from_tag_right,
+                         std::string to_tag_left, std::string to_tag_right );
```

Comment:
If the string contains `[color_XXX]` (not likely but no impossible) this will mangle the string. It might be safer to skip tags in the form of `<color_xxx>` and `</color>` when expanding the tags in `SNIPPET.expand` and `parse_tags`.

Reply:
At times, another scenario is possible as well:
a global variable color has a string value "<color_red>"

In such a case, the parsed global_val needs to be parsed again for its color tags.
Therefore, I choose to handle this within the loop that parses tags.

Perhaps I can replace [] with some less common characters, such as 【】?

Reply:
> At times, another scenario is possible as well:
> a global variable color has a string value "<color_red>"

I think it will not be any different from normal text as long as we ignore color tags for the expansion purpose. Color tags are handled by the printing functions rather than the expansion functions.

Reply:
In the process of searching for "<" and ">", I skipped the color tag. 

Since the color tag is enclosed with "<" and ">", preprocessing the string with `Snippet.expand` does not seem to transform it into another form that does not interfere with the detection of "<" and ">".